### PR TITLE
Catching a nil pointer error

### PR DESCRIPTION
**Pull request by bigkraig**

### CODE REVIEW

**Comment by bigkraig** [comment]
@M00nF1sh

File: internal/aws/albrgt/rgt.go [hunk 1]
```diff
@@ -147,6 +147,35 @@ func (r *RGT) GetClusterResources() (*Resources, error) {
 		}
 	}
 
+	// Legacy deployments may not have the proper tags, and RGT doesn't allow you to use wildcards on names
+	p := request.Pagination{
+		EndPageOnSameToken: true,
+		NewRequest: func() (*request.Request, error) {
+			req, _ := r.GetResourcesRequest(&resourcegroupstaggingapi.GetResourcesInput{
```
Comment (cv) [inline internal/aws/albrgt/rgt.go]: Couldn't this use `GetResourcesPages` too, like that call we changed a couple of days ago in #605?

Comment (bigkraig) [reply internal/aws/albrgt/rgt.go]: Yeah we can. I went with the old method since we hadn’t merged that yet.

File: pkg/util/types/tags.go [hunk 2]
```diff
@@ -64,14 +64,17 @@ func (t ELBv2Tags) ServiceNameAndPort() (name string, port intstr.IntOrString, e
...
+
+	if name == "" {
 		return "", intstr.IntOrString{}, fmt.Errorf("kubernetes.io/service-name tag is missing")
 	}
 
 	if v, ok := t.Get("kubernetes.io/service-port"); ok {
 		port = intstr.Parse(v)
 	} else {
-		return "", intstr.IntOrString{}, fmt.Errorf("kubernetes.io/service-port is missing")
+		port = intstr.Parse("1")
```
Comment (M00nF1sh) [inline pkg/util/types/tags.go]: will this be an issue? i think it's better to keep the original one..

Comment (bigkraig) [reply pkg/util/types/tags.go]: It was necessary to upgrade from an old version of the controller that didnt have the `kubernetes.io/service-port` tag yet. TBH now that we are using TG port 1, i need to look and see if this tag is still useful

Comment (M00nF1sh) [reply pkg/util/types/tags.go]: the service-port is what as specified in ingress spec, e.g.(80/http), it's not same as the target port(port on pods) 😄

Comment (bigkraig) [reply pkg/util/types/tags.go]: so the tag is still useful but if we come across a TG w/o it, we shouldn't abort the assembly by returning an error here. perhaps a `intstr.Parse("0")` to identify it as an invalid TG and force a recreation of it?

Comment (M00nF1sh) [reply pkg/util/types/tags.go]: i think it’s fine to use port0, since the worse case is the targetgroup will be orphaned :smile:

**Pull request merged**